gsdk/src/storage.rs

```diff
@@ -286,6 +286,26 @@ impl Api {
 
         Ok(mailbox)
     }
+
+    /// Get up to `count` messages from the mailbox
+    pub async fn all_mailbox(&self, count: u32) -> Result<Vec<(StoredMessage, Interval<u32>)>> {
```

Comment:
This method is too similar to the `mailbox` method. Better would be to generalize them two.
```suggestion
    /// Get up to `count` messages from the mailbox.
    pub async fn all_mailbox(&self, count: u32) -> Result<Vec<(StoredMessage, Interval<u32>)>> {
```